src/main/java/com/mixpanel/android/mpmetrics/DecideMessages.java

```diff
@@ -104,6 +104,13 @@ public synchronized void reportResults(List<Survey> newSurveys, List<InAppNotifi
             }
         }
 
+        // in the case we do not receive a new variant, this means the A/B test should be turned off
+        if(!hasNewVariants && newVariantsLength == 0) {
```

Comment:
How does this solve the problem? Say we have `variantA` and `variantB` running. If we turn off `variantB`, this logic will  only work if a) we do not add a `variantC` and b) we also remove `variantA`.